### PR TITLE
[Loot] Fix issue with nested data being loaded multiple times

### PR DESCRIPTION
**Pull request by Akkadius**
Fixed an issue where nested data was getting loaded into memory multiple times. Added checks in nested loot data to ensure that nothing gets double added to memory.

### CODE REVIEW

**Comment by fryguy503** [comment]
So far this is looking massively better!

File: zone/zone.h [hunk 1]
```diff
@@ -437,7 +437,7 @@ class Zone {
 
 	// loot
 	void LoadLootTable(const uint32 loottable_id);
-	void LoadLootTables(const std::vector<uint32>& loottable_ids);
```
Comment (KimLS) [inline zone/zone.h:440]: Why did we remove this?

File: zone/zone_loot.cpp [hunk 2]
```diff
@@ -5,20 +5,36 @@
 #include "../common/repositories/lootdrop_repository.h"
 #include "../common/repositories/lootdrop_entries_repository.h"
 
-void Zone::LoadLootTables(const std::vector<uint32> &loottable_ids)
```
Comment (KimLS) [inline zone/zone_loot.cpp:8]: And this reference, why did we remove it?

**Review by KimLS — CHANGES_REQUESTED** [review]
Don't see any issues otherwise but those dropped references forces a copy for no apparent reason.

File: zone/zone_loot.cpp (hunk 2 above)
Comment (Akkadius) [reply zone/zone_loot.cpp:8]: We modify it by filtering on it below, it wouldn't work on the reference so I made a copy. Very low overhead operation to do the copy anyhow

File: zone/zone.h (hunk 1 above)
Comment (Akkadius) [reply zone/zone.h:440]: Same reference as below, answered

**Review by KimLS — APPROVED** [review]

**Comment by Akkadius** [comment]
We spoke on this one

**Pull request merged**